src/main/java/duke/task/TaskManager.java

```diff
@@ -0,0 +1,181 @@
+            System.out.println("\tNo such task");
+            return;
+        }
+        Todo task = tasks.get(taskNumber - 1);
+        tasks.remove(taskNumber-1);
+        System.out.println("\tNoted. I've removed this task:");
+        System.out.println("\t\t"+task.toString());
+        System.out.println("\t Now you have "+tasks.size()+" tasks in the list.");
+        updateFile();
+    }
+
+    private static void printTaskAddedConfirmation(Todo task) {
+        System.out.println("\tGot it. I've added this task:");
+        System.out.println("\t\t" + task);
+        System.out.format("\tNow you have %d tasks in the list.\n", tasks.size());
+    }
+
+    private static void updateFile(){
+        try {
+            FileIOManager.write(tasks);
+        } catch (IOException e){
+            System.out.println(e);
+        }
+    }
+    public static void convertTaskStringToTasks(ArrayList<String> tasksString){
+        if (tasksString==null){
+            return;
+        }
+        if (tasksString.isEmpty()){
+            return;
+        }
+        int startIndexOfTask;
+        int endIndexOfTask;
+        int startIndexOfDate;
+        int endIndexOfDate;
+        String date;
+        String taskName;
+        for (String taskString: tasksString){
+            char type = taskString.charAt(3);
+            switch(type) {
+            case 'T':
+                startIndexOfTask = 9;
+                taskName = taskString.substring(startIndexOfTask);
+                Todo newTodo = new Todo(taskName);
+                tasks.add(newTodo);
+                if (taskString.charAt(6)=='X'){
+                    newTodo.setIsDone();
+                }
+                continue;
+            case 'D':
+                startIndexOfTask = 9;
+                endIndexOfTask = taskString.indexOf('(')-1;
+                startIndexOfDate = endIndexOfTask+6;
+                endIndexOfDate = taskString.indexOf(')');
+                taskName = taskString.substring(startIndexOfTask, endIndexOfTask);
+                date = taskString.substring(startIndexOfDate,endIndexOfDate);
+                Deadline newDeadline = new Deadline(taskName, date);
+                tasks.add(newDeadline);
+                if (taskString.charAt(6)=='X'){
+                    newDeadline.setIsDone();
+                }
+                continue;
+            case'E':
+                startIndexOfTask = 9;
+                endIndexOfTask = taskString.indexOf('(')-1;
+                startIndexOfDate = endIndexOfTask+6;
+                endIndexOfDate = taskString.indexOf(')');
+                taskName = taskString.substring(startIndexOfTask, endIndexOfTask);
+                date = taskString.substring(startIndexOfDate,endIndexOfDate);
+                Event newEvent = new Event(taskName, date);
+                tasks.add(newEvent);
+                if (taskString.charAt(6)=='X'){
+                    newEvent.setIsDone();
+                }
```

Comment:
This logic can easily be abstracted out into a separate function. The reason being you don't want to repeat yourself and you don't want to make the code extremely long and hard to read. You can also consider comments to improve your code readability.